From de2c73d27693c86051753ee5aa4b95a4bf05ff97 Mon Sep 17 00:00:00 2001
From: QbProg <tholag@gmail.com>
Date: Sat, 19 May 2012 15:54:42 +0200
Subject: [PATCH] [warning-fix][uninitialized-var] Fixes warnings like :
 (potentially) uninitialized local variable 'name' used

A return for the "default" switch case was added, since all the vars
where used uninitialized in that case. Shouldn't happen anyway.
---
 src/AdvApp2Var/AdvApp2Var_Iso.cxx |    1 +
 1 file changed, 1 insertion(+)

Index: oce/src/AdvApp2Var/AdvApp2Var_Iso.cxx
===================================================================
--- oce.orig/src/AdvApp2Var/AdvApp2Var_Iso.cxx
+++ oce/src/AdvApp2Var/AdvApp2Var_Iso.cxx
@@ -174,6 +174,7 @@ void AdvApp2Var_Iso::MakeApprox(const Ad
   case GeomAbs_NoneIso:
     //pkv t
   default:
+	  return;
     break;
     //#endif
   }
Index: oce/src/AlienImage/AlienImage_BMPAlienData.cxx
===================================================================
--- oce.orig/src/AlienImage/AlienImage_BMPAlienData.cxx
+++ oce/src/AlienImage/AlienImage_BMPAlienData.cxx
@@ -89,7 +89,7 @@ Standard_Boolean AlienImage_BMPAlienData
   int   isInsideOut, hasMask, nBitCount;
   int   iCompression, isOS2Format, hasColormap;
   int   nBytes, iDataSize;
-  char  bmpSign[2];
+  char  bmpSign[2] = {'\0' , '\0'};
   AlienImage_BMPHeader bmfh; // Standard BMP file header
   BITMAPINFOHEADER bmih = {}; // Windows BMP header
   BITMAPCOREHEADER bmch = {}; // OS/2 BMP header
Index: oce/src/BOPAlgo/BOPAlgo_BOP.cxx
===================================================================
--- oce.orig/src/BOPAlgo/BOPAlgo_BOP.cxx
+++ oce/src/BOPAlgo/BOPAlgo_BOP.cxx
@@ -467,7 +467,7 @@ static
   void BOPAlgo_BOP::BuildRC()
 {
   Standard_Boolean bFlag;
-  Standard_Integer i, aDmin, aNb[2], iX, iY;
+  Standard_Integer i, aDmin, aNb[2], iX = 0, iY = 0;
   TopAbs_ShapeEnum aTmin;
   TopoDS_Compound aC, aCS[2];
   BRep_Builder aBB;
Index: oce/src/BSplCLib/BSplCLib_2.cxx
===================================================================
--- oce.orig/src/BSplCLib/BSplCLib_2.cxx
+++ oce/src/BSplCLib/BSplCLib_2.cxx
@@ -589,14 +589,14 @@ void BSplCLib::MovePointAndTangent(const
   conditions,
   start_num_poles,
   end_num_poles,
-  index,
+  index = 0,
   start_index,
   end_index,
   other_index,
   type,
   order ;
   
-  Standard_Real    new_parameter,
+  Standard_Real    new_parameter = 0.0,
   value,
   divide,
   end_value,
Index: oce/src/LDOM/LDOM_BasicElement.cxx
===================================================================
--- oce.orig/src/LDOM/LDOM_BasicElement.cxx
+++ oce/src/LDOM/LDOM_BasicElement.cxx
@@ -223,7 +223,7 @@ const LDOM_BasicNode * LDOM_BasicElement
                                    const LDOM_BasicNode           * aLastCh)
 {
   //  Create attribute
-  Standard_Integer aHash;
+  Standard_Integer aHash = 0;
   LDOM_BasicAttribute& anAttr =
     LDOM_BasicAttribute::Create (anAttrName, aDocument, aHash);
   anAttr.myValue = anAttrValue;
@@ -453,7 +453,7 @@ void LDOM_BasicElement::ReplaceElement
 loop_attr:
   LDOM_BasicNode * aLastAttr = (LDOM_BasicNode *) aLastChild;
   for (; aBNode != NULL; aBNode = aBNode -> GetSibling()) {
-    Standard_Integer aHash;
+    Standard_Integer aHash = 0;
     if (aBNode -> isNull()) continue;
     const LDOM_BasicAttribute * aBNodeAtt= (const LDOM_BasicAttribute *) aBNode;
     LDOM_BasicAttribute * aNewAtt =
Index: oce/src/ViewerTest/ViewerTest.cxx
===================================================================
--- oce.orig/src/ViewerTest/ViewerTest.cxx
+++ oce/src/ViewerTest/ViewerTest.cxx
@@ -626,8 +626,8 @@ static int visos (Draw_Interpretor& di,
 
   Standard_Integer aLastInd = argc - 1;
   Standard_Boolean isChanged = Standard_False;
-  Standard_Integer aNbUIsos;
-  Standard_Integer aNbVIsos;
+  Standard_Integer aNbUIsos = 0;
+  Standard_Integer aNbVIsos = 0;
 
   if (aLastInd >= 3) {
     Standard_Boolean isIsoOnPlane = Standard_False;
